flow/shared/exceptions/mongo.go

```diff
@@ -0,0 +1,6 @@
+package exceptions
+
+type MongoRetryablePoolError interface {
+	error
+	Retryable() bool
```

Comment:
unfortunately this will catch all errors that implement this rather genericly named method

fix is to do one of:
1. make this something we'll generally accept, but then don't tag as mongo
2. use reflect to make sure the error we get is from mongo type

Reply:
i like 1 :)